src/link.js

```diff
@@ -178,40 +144,80 @@ export default class Link extends Feature {
 		const balloonPanel = new LinkBalloonPanel( panelModel, new LinkBalloonPanelView( editor.locale ) );
 
 		// Observe `LinkBalloonPanelMode#executeLink` event from within the model of the panel,
-		// which means that the `Save` button has been clicked.
+		// which means that form has been submitted.
 		this.listenTo( panelModel, 'executeLink', () => {
 			editor.execute( 'link', balloonPanel.urlInput.value );
-			balloonPanel.view.hide();
+			this._hidePanel( { focusEditable: true } );
 		} );
 
 		// Observe `LinkBalloonPanelMode#executeUnlink` event from within the model of the panel,
 		// which means that the `Unlink` button has been clicked.
 		this.listenTo( panelModel, 'executeUnlink', () => {
 			editor.execute( 'unlink' );
-			balloonPanel.view.hide();
+			this._hidePanel( { focusEditable: true } );
+		} );
+
+		// Observe `LinkBalloonPanelMode#executeCancel` event from within the model of the panel,
+		// which means that the `Cancel` button has been clicked.
+		this.listenTo( panelModel, 'executeCancel', () => this._hidePanel( { focusEditable: true } ) );
+
+		// Handle `Ctrl+L` keystroke and show panel.
+		editor.keystrokes.set( 'CTRL+L', () => {
+			this._attachPanelToElement();
+			balloonPanel.urlInput.view.select();
 		} );
 
-		// Always focus editor on panel hide.
+		// Attach close by `Esc` press and click out of panel actions on panel show, on panel hide clean up listeners.
 		this.listenTo( balloonPanel.view.model, 'change:isVisible', ( evt, propertyName, value ) => {
-			if ( !value ) {
-				viewDocument.focus();
+			if ( value ) {
+				// Handle close by `Esc`.
+				balloonPanel.view.listenTo( document, 'keydown', this._closePanelOnClick.bind( this ) );
+
+				// Handle close by clicking out of the panel.
+				// Note that it is not handled by a `click` event, it is because clicking on link button or directly on link element
+				// was opening and closing panel at the same time.
+				balloonPanel.view.listenTo( document, 'mouseup', ( evt, domEvt ) => {
+					// Do nothing when the panel was clicked.
+					if ( balloonPanel.view.element.contains( domEvt.target ) ) {
+						return;
+					}
+
+					// When click was out of the panel then hide it.
+					balloonPanel.view.hide();
+
+					// When editor was clicked then restore editor focus.
+					if ( editor.ui.view.element.contains( domEvt.target ) ) {
```

Comment:
It is bad and it is not tested. I know that features shouldn't depend on `EditorUI`, but we need to figure out how to restore focus on editable after clicking on some part of EditorUI e.g. toolbar button. Maybe EditorUI should do it itself? Listen to click on itself, ask `FocusManager` where is the focus, and restore if is needed? Or maybe there is an easier way which I don't see.